client/src/components/menubar/index.tsx

```diff
@@ -84,9 +90,53 @@ interface State {
     clipPercentileMax: number | undefined;
   };
 }
-class MenuBar extends React.PureComponent<MenuBarProps, State> {
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  static isValidDigitKeyEvent(e: any) {
+
+const DIFFEXP_WIDTH = 150;
+const BUTTON_WIDTH = 38;
+
+function MenuBar(props: MenuBarProps) {
+  const {
+    dispatch,
+    disableDiffexp,
+    clipPercentileMin,
+    clipPercentileMax,
+    graphInteractionMode,
+    showCentroidLabels,
+    categoricalSelection,
+    colorAccessor,
+    subsetPossible,
+    subsetResetPossible,
+    screenCap,
+    imageUnderlay,
+  } = props;
+
+  const [pendingClipPercentiles, setPendingClipPercentiles] = useState<
+    State["pendingClipPercentiles"]
+  >({
+    clipPercentileMin: undefined,
+    clipPercentileMax: undefined,
+  });
+
+  const [widthUnits, setWidthUnits] = useState(8);
+
+  const ref = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    const observer = new ResizeObserver((rightMenuBar) => {
+      setWidthUnits(
+        (rightMenuBar[0].contentRect.width - DIFFEXP_WIDTH) / BUTTON_WIDTH
+      );
+    });
+
+    if (ref.current) {
+      observer.observe(ref.current);
+    }
+
+    return () => {
+      observer.disconnect();
+    };
+  }, []);
```

Comment:
New state for handling width changes to menubar. Use a plain width unit to not have to deal with pixels.